### PR TITLE
oh-my-posh3: Add version 3.43.2

### PR DESCRIPTION
**Pull request by Vixb1122**

### CODE REVIEW

**Review by Ash258 — CHANGES_REQUESTED** [review]
Completely remove everything you have done and implement manifest according to the official docs, include credits to original author.

https://ohmyposh.dev/docs/scoop/

**Comment by Vixb1122** [comment]
@Ash258 fixed

**Review by Ash258 — CHANGES_REQUESTED** [review]
... `git mv bucket/oh-my-posh3 bucket/oh-my-posh3.json`

**Comment by Vixb1122** [comment]
@Ash258 AHa, sorry

File: bucket/oh-my-posh3.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        "oh-my-posh-wsl",
+        "oh-my-posh.exe"
+    ],
+    "checkver": {
+        "github": "https://github.com/JanDeDobbeleer/oh-my-posh3"
+    },
+    "post_install": [
+        "Write-Host 'Thanks for installing Oh my Posh.'",
+        "Write-Host 'Have a look at https://ohmyposh.dev/docs/installation for detailed instructions for your shell.'"
+    ],
```
Comment (Ash258) [inline bucket/oh-my-posh3.json:25]:
```suggestion
```

File: bucket/oh-my-posh3.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{
+    "version": "3.43.1",
+    "description": "A prompt theme engine for any shell",
+    "homepage": "https://ohmyposh.dev",
+    "license": {
+        "identifier": "GPL-3.0-only",
+        "url": "https://github.com/JanDeDobbeleer/oh-my-posh3/blob/main/COPYING"
+    },
```
Comment (Ash258) [inline bucket/oh-my-posh3.json:8]:
```suggestion
    "license": "GPL-3.0-only",
    "notes": "Refer to 'https://ohmyposh.dev/docs/configure' for shell specific configurations.",
```

File: bucket/oh-my-posh3.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    "architecture": {
+        "64bit": {
+            "url": "https://github.com/JanDeDobbeleer/oh-my-posh3/releases/download/v3.43.1/posh-windows-wsl-amd64.7z",
+            "hash": "1AEFCC4BEC99A840B87491E2ABEC37F9CD743A077331810E7BFBA86DD6E096C5"
```
Comment (Ash258) [inline bucket/oh-my-posh3.json:12]:
```suggestion
            "hash": "1aefcc4bec99a840b87491e2abec37f9cd743a077331810e7bfba86dd6e096c5"
```

File: bucket/oh-my-posh3.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    ],
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://github.com/JanDeDobbeleer/oh-my-posh3/releases/download/v$version/posh-windows-wsl-amd64.7z",
+                "hash": {
+                    "url": "$url.sha256",
+                    "find": "([a-fA-F0-9]{64})"
+                }
+            }
```
Comment (Ash258) [inline bucket/oh-my-posh3.json:34]:
```suggestion
                "url": "https://github.com/JanDeDobbeleer/oh-my-posh3/releases/download/v$version/posh-windows-wsl-amd64.7z"
            }
        },
        "hash": {
            "url": "$url.sha256"
```

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/oh-my-posh3.json (hunk 3 above)
Comment (Vixb1122) [reply bucket/oh-my-posh3.json:12]: this hash is generate by [`$zipHash = Get-FileHash ./posh-windows-wsl-amd64.7z -Algorithm SHA256`](https://github.com/JanDeDobbeleer/oh-my-posh3/blob/2a54724a7514a8d4fcbeed2f055df5e8b11fad3b/packages/scoop/build.ps1#L22) and get uppercase result.
Next version by autoupdate will be uppercase again.

Comment (Ash258) [reply bucket/oh-my-posh3.json]: Stop deleting comments...

You have no idea what you are talking about. The hash is extracted in autoupdate, which extract the hash from the specified and then execute `format_hash`, which return `$hash.ToLowe()`

Comment (Vixb1122) [reply bucket/oh-my-posh3.json]: Sorry, I realized that I understood what you meant after performing the autoupdate, so I deleted the comment(sorry).

Comment (Vixb1122) [reply bucket/oh-my-posh3.json]: I find that i use a 'checkver.ps1' from https://github.com/Ash258/GenericBucket that without feature about `$hash.ToLower()`.
Can you update scripts from [GenericBucket](https://github.com/Ash258/GenericBucket) to sync with this repo. @Ash258

Comment (Ash258) [reply bucket/oh-my-posh3.json]: It is coming from Core not bucket specific scripts.

Comment (Ash258) [reply bucket/oh-my-posh3.json]: https://github.com/lukesampson/scoop/blob/643cfb0da8369dbbe2964511698763c5ff179492/lib/autoupdate.ps1#L89
https://github.com/lukesampson/scoop/blob/643cfb0da8369dbbe2964511698763c5ff179492/lib/core.ps1#L856-L867

Comment (Ash258) [reply bucket/oh-my-posh3.json]: I will not sync anything as they are exactly functinaly equal scripts...

Comment (Vixb1122) [reply bucket/oh-my-posh3.json]: okay, thx.

**Comment by Vixb1122** [comment]
Please review again.

File: bucket/oh-my-posh3.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{
+    "version": "3.43.1",
+    "description": "A prompt theme engine for any shell",
+    "homepage": "https://ohmyposh.dev",
+    "license": "GPL-3.0-only",
+    "notes": "Refer to 'https://ohmyposh.dev/docs/configure' for shell specific configurations.",
```
Comment (Ash258) [inline bucket/oh-my-posh3.json:6]:
```suggestion
    "notes": "Refer to 'https://ohmyposh.dev/docs/installation#4-replace-your-existing-prompt' for shell specific configurations.",
```

File: bucket/oh-my-posh3.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    "architecture": {
+        "64bit": {
+            "url": "https://github.com/JanDeDobbeleer/oh-my-posh3/releases/download/v3.43.1/posh-windows-wsl-amd64.7z",
+            "hash": "1aefcc4bec99a840b87491e2abec37f9cd743a077331810e7bfba86dd6e096c5"
+        }
+    },
+    "bin": [
+        "oh-my-posh-wsl",
+        "oh-my-posh.exe"
```
Comment (Ash258) [inline bucket/oh-my-posh3.json:15]:
```suggestion
        "oh-my-posh.exe",
        "oh-my-posh-wsl"
```

File: bucket/oh-my-posh3.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    ],
+    "checkver": {
```
Comment (Ash258) [inline bucket/oh-my-posh3.json:17]:
```suggestion
    "persist": "themes",
    "checkver": {
```

**Review by Ash258 — CHANGES_REQUESTED** [review]

File: bucket/oh-my-posh3.json (hunk 6 above)
Comment (Vixb1122) [reply bucket/oh-my-posh3.json:15]: Is there a potential problem here?

**Comment by Vixb1122** [comment]
re verify please

**Comment by Ash258** [comment]
Move this PR into main bucket with following commit message

```
oh-my-posh3: Add version 3.43.2

Co-authored-by: Jan De Dobbeleer <jan.de.dobbeleer@gmail.com>
Co-authored-by: Ash258 <cabera.jakub@gmail.com>
```

**Pull request closed without merging**